src/key-value-engine/B-Tree_vs_The_Log-Structured_Merge-Tree.md

```diff
@@ -0,0 +1,101 @@
+
+LSM-tree performs `compaction`  to merges several sstables into one new sstable which contains only the live data from the input sstables. Compation helps LSM-tree to recycle space and reduce read amplification.  There are two kinds of `compaction strategy` which is `Size-Tiered compaction strategy` (STCS) and `Level-Based compaction strategy` (LBCS). The idea behind STCS is compacting small sstables into medium sstables when LSM-tree has enough small sstables and compacting medium sstables into large sstables when LSM-tree has enough medium sstables. The idea of LBCS is to organize data into levels which contains one sorted run.
+
+This section is going to discuss about the write amplification and read amplification of B+tree and Level-Based LSM-tree. 
+
+### B+ Tree
+
+In the B+ tree, copies of the keys are stored in the internal nodes; the keys and records are stored in leaves; in addition, a leaf node may include a pointer to the next leaf node to speed sequential access.
+
+To simplify the analysis, let's say that the block size of the tree as `B`, and assumes that keys, pointers, and records are constant sized, so that each internal node contains `O(B)` children and each leaf contains `O(B)` data records. (The root node is a special case, and can be nearly empty in some situations.) Under all these assumptions, the depth of a B tree is 
+$$
+O(log_BN/B)
+$$
+where `N` is the size of database.
+
+#### Write Amplification
+
+For the worst-case insertion workloads, every insertion requires writing the leaf block containing the record, so the write amplification is `B`.
+
+#### Read Amplification
+
+The number of I/O's for any query is at most `O(logB N/B)` which is the depth of the tree. 
+
+### Level-Based LSM-tree
+
+In level-based LSM-tree, data is organized into levels. Each level contains one run. Data starts in level 0, then gets merged into the level 1 run. Eventually the level 1 run is merged into the level 2 run, and so forth. Each level is constrained in its sizes. Growth factor `k`  is specified as the magnification of data size at each level 
+$$
+level_i = level_{i-1}*k
+$$
+We can analyze level-based LSM-tree as follows. If the growth factor is `k` and the smallest level is a single file of size `B`, then the number of levels is 
+
+$$
+Θ(log_kN/B)
+$$
+
+#### Write Amplification
+
+Data must be moved out of each level once, but data from a given level is merged repeatedly with data from the previous level. On average, after being first written into a level, each data item is remerged back into the same level about `k/2` times. So the total write amplification is 
+$$
+Θ(k*log_kN/B)
+$$
+
+#### Read Amplification
+
+To perform a short range query in the cold cache case, we must perform a binary search
+on each of the levels.
+
+For the highest level i, the data size is O(N), so that it performs O(log N/B)disk reads. 
```

Comment:
I recommend wrapping all time-complexities in a code block like you did above (to be more consistent)